tm-bench/main.go

```diff
@@ -119,19 +120,38 @@ func latestBlockHeight(client tmrpc.Client) int64 {
 	return status.SyncInfo.LatestBlockHeight
 }
 
+// Calculates the tx / second, and blocks / second based off of the number the transactions
+// and number of blocks that occured from the start block, and the end time.
 func calculateStatistics(client tmrpc.Client, minHeight int64, timeStart, timeStop time.Time, duration int) *statistics {
 	stats := &statistics{
 		BlocksThroughput: metrics.NewHistogram(metrics.NewUniformSample(1000)),
 		TxsThroughput:    metrics.NewHistogram(metrics.NewUniformSample(1000)),
 	}
 
 	// get blocks between minHeight and last height
+	// This returns max(minHeight,(last_height - 20)) to last_height
 	info, err := client.BlockchainInfo(minHeight, 0)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 
+	lastHeight := info.LastHeight
+	diff := lastHeight - minHeight
+
+	blockMetas := info.BlockMetas
+	offset := len(blockMetas)
+	for len(blockMetas) < int(diff) {
+		// get blocks between minHeight and last height
+		info, err := client.BlockchainInfo(minHeight, lastHeight-int64(offset))
```

Comment:
don't we need to progress the minHeight forward as well ?

Reply:
I don't think we have to, since the block at the minheight isn't getting returned until lastHeight is within 20 blocks of minHeight. I'll test a couple more times to ensure that its printing fully in order though.

Reply:
oh I see, it counts down from the maxHeight parameter